extension/src/popup/components/sendPayment/SendConfirm/TransactionDetails/index.tsx

```diff
@@ -244,8 +244,11 @@ export const TransactionDetails = ({ goBack }: { goBack: () => void }) => {
       (tag) => tag === TRANSACTION_WARNING.memoRequired && !memo,
     );
 
-  const isAssetMalicious =
-    accountBalances.balances?.[asset].isMalicious || false;
+  let isAssetMalicious = accountBalances.balances?.[asset].isMalicious || false;
+  if (!isAssetMalicious && destinationAsset) {
+    isAssetMalicious =
+      accountBalances.balances?.[destinationAsset].isMalicious || false;
+  }
```

Comment:
This makes sure to check the destination asset (in a swap) to see if the user is swapping to a malicious asset